using_restricted_array.rb

```diff
@@ -7,41 +7,117 @@
 # Calculates the length of the restricted array. All values are integers.
 # The restricted_array is terminated by 'nil' i.e. array[length] = nil
 def length(array)
-  puts "NOT IMPLEMENTED"
+  index = 0
+  element = 0
+  while element.class == Integer
+    element = array[index]
+    index += 1
+  end
+  index - 1
 end
 
+
 # Prints each integer values in the array
 def print_array(array)
-  puts "NOT IMPLEMENTED"
+  index = 0
+  element = 0
+  while element.class == Integer
+    element = array[index]
+    index += 1
+    print "#{element} "
+  end
+  puts
+  index - 1
 end
 
+
 # For an unsorted array, searches for 'value_to_find'.
 # Returns true if found, false otherwise.
 def search(array, length, value_to_find)
-  puts "NOT IMPLEMENTED"
+  element = nil
+  index = 0
+  while element != value_to_find
+    element = array[index]
+    index += 1
+    return false if index > length
+  end
+  return index - 1
```

Comment:
You should be returning `true` or `false` from this method. I'd recommend changing line 44 to explicitly return `true` instead of leveraging Ruby specific truthiness.